$OpenBSD: patch-snmplib_read_config_c,v 1.3 2016/10/12 10:50:33 sthen Exp $

commit 477b4307ef12ddce3b6a9205e0bdddbfb2e0e9b6
Author: Bart Van Assche <bvanassche@acm.org>
Date:   Sat Feb 20 18:58:18 2016 -0800

    Remove U64 typedef

    The U64 typedef conflicts with a typedef in a Perl header file. Hence
    remove the U64 typedef from the Net-SNMP header files. This patch does
    not modify the Net-SNMP ABI.

--- snmplib/read_config.c.orig	Mon Dec  8 20:23:22 2014
+++ snmplib/read_config.c	Wed Oct 12 10:50:36 2016
@@ -2270,10 +2270,10 @@ read_config_read_memory(int type, char *readfrom,
         return readfrom;
 
     case ASN_COUNTER64:
-        if (*len < sizeof(U64))
+        if (*len < sizeof(struct counter64))
             return NULL;
-        *len = sizeof(U64);
-        read64((U64 *) dataptr, readfrom);
+        *len = sizeof(struct counter64);
+        read64((struct counter64 *) dataptr, readfrom);
         readfrom = skip_token(readfrom);
         return readfrom;
     }
